packages/portal-proto/src/features/facets/FilterPanel.tsx

```diff
@@ -0,0 +1,152 @@
+ * @param showReset - whether to show the reset filters to default option
+ * @param handleClearAll - callback to remove custom filters
+ * @param hideIfEmpty - hide facets if they do not have data
+ * @param showPercent - whether to show the count percent of whole
+ * @param isLoading - whether the filter defintions are loading
+ */
+
+const FilterPanel = ({
+  facetDefinitions,
+  facetHooks,
+  valueLabel,
+  app,
+  toggleAllFiltersExpanded,
+  allFiltersCollapsed,
+  hasCustomFilters = false,
+  handleRemoveFilter = undefined,
+  handleCustomFilterSelected = undefined,
+  getDefaultFacets = undefined,
+  customFacetsConfig = undefined,
+  showReset = false,
+  handleClearAll = undefined,
+  hideIfEmpty = false,
+  showPercent = true,
+  isLoading = false,
+}: FilterPanelProps) => {
+  const [opened, setOpened] = useState(false);
+
+  return (
+    <div className="w-1/3 xl:w-1/4 pl-[calc(100vw - 100%)]">
+      <div className="flex flex-col gap-y-3 mr-3 mb-4">
+        <div>
+          <Text size="lg" className="text-primary-content-darker font-bold">
+            Filters
+          </Text>
+        </div>
+        <div className="flex justify-between">
+          <button onClick={() => toggleAllFiltersExpanded(allFiltersCollapsed)}>
+            {allFiltersCollapsed ? "Expand All" : "Collapse All"}
+          </button>
+          {showReset && (
+            <button onClick={() => handleClearAll()}>Reset Filters</button>
+          )}
+        </div>
+        {hasCustomFilters && (
+          <>
+            <Button
+              variant="outline"
+              aria-label="Add a custom filter"
+              data-testid="button-add-a-file-filter"
+              className="flex justify-center items-center border-primary-darker text-primary hover:text-base-max hover:bg-primary rounded-md"
+              onClick={() => setOpened(true)}
+            >
+              <AddAdditionalIcon className="text-2xl xl:text-xl" />
+              <Text className="text-sm font-bold">Add a Custom Filter</Text>
+            </Button>
+            <div className="flex flex-col gap-y-4" data-testid="filters-facets">
+              <Modal
+                data-testid="modal-repository-add-custom-filter"
+                size="xl"
+                opened={opened}
+                onClose={() => setOpened(false)}
+                title="Add a Custom Filter"
+              >
+                <div className="p-4">
+                  <FacetSelection
+                    facetType="files"
+                    handleFilterSelected={(filter: string) => {
+                      handleCustomFilterSelected(filter);
+                      setOpened(false);
+                    }}
+                    usedFacets={customFacetsConfig}
+                  />
+                </div>
+              </Modal>
+            </div>
+          </>
+        )}
+        <LoadingOverlay data-testid="loading-spinner" visible={isLoading} />
```

Comment:
Will we need position `relative` for its parent to limit the loading overlay be displayed within its parent area? If so, I don't see it.

Reply:
Good question. This loading overlay was something that the repository filter panel had but I'm not even sure when it is triggered. I'll look into it more.

Reply:
I couldn't figure out a scenario where this loading overlay actually happens even with throttled speeds/incognito mode so I removed it 🤷‍♀️

Reply:
Yes, let's just YOLO it! 😄